CMakeLists.txt

```diff
@@ -70,18 +70,27 @@ else()
         GIT_REPOSITORY https://github.com/apple/swift-foundation-icu.git
         GIT_TAG 0.0.9)
 endif()
+FetchContent_MakeAvailable(SwiftFoundationICU)
 
-if (_SwiftCollections_SourceDIR)
-    message(STATUS "_SwiftCollections_SourceDIR provided, using swift-foundation-icu checkout at ${_SwiftCollections_SourceDIR}")
-    FetchContent_Declare(SwiftCollections
-        SOURCE_DIR ${_SwiftCollections_SourceDIR})
-else()
-    message(STATUS "_SwiftCollections_SourceDIR not provided, checking out local copy of swift-collections")
-    FetchContent_Declare(SwiftCollections
-        GIT_REPOSITORY https://github.com/apple/swift-collections.git
-        GIT_TAG 1.1.2)
-endif()
-FetchContent_MakeAvailable(SwiftFoundationICU SwiftCollections)
+block(SCOPE_FOR POLICIES)
+    # swift-collections does not build properly without swift-driver, which
+    # may not be available during the toolchain build. Set CMP0157 to OLD to
+    # work around this issue.
+    if(POLICY CMP0157)
+        cmake_policy(SET CMP0157 OLD)
```

Comment:
If this only applies to swift-collections, can we make this change to swift-collections instead of to swift-foundation? There are other things in the toolchain that use swift-collections and I think it'd be best to fix it at the source rather than fixing it in each project that uses swift-collections in the toolchain (even if swift-foundation is the only project this affects, it'd be best to fix this in swift-collections since that's where this policy is actually set). cc @lorentey